src/Categories/Multi/Category/Cartesian.agda

```diff
@@ -0,0 +1,115 @@
+{-# OPTIONS --without-K --safe #-}
+
+-- Idea:
+-- Whereas ordinary multicategories are based around the notion of substitution
+-- from noncommutative linear (Lambek) logic, Cartesian multicategories are
+-- based around the notion of substitution from intuitionistic logic, which is
+-- a lot simpler. A representable Cartesian multicategory is exactly a
+-- Cartesian category.
+--
+-- See https://ncatlab.org/nlab/show/cartesian+multicategory, particularly the
+-- section “Alternative presentation”
+-- (https://ncatlab.org/nlab/show/cartesian+multicategory#alternative_presentation).
+
+module Categories.Multi.Category.Cartesian where
+
+open import Data.List hiding ([_])
+open import Data.List.Membership.Propositional
+open import Data.Wrap
+open import Level
+open import Relation.Binary using (Rel; Setoid; IsEquivalence)
+import Relation.Binary.Reasoning.Setoid as SetoidR
+
+infix 4 [_]_≈ᵉ_
+
+Env : ∀ {o ℓ} {X : Set o} (V : X → Set ℓ) (Δ : List X) → Set (o ⊔ ℓ)
```

Comment:
I think this file could do with a lot more comments, giving a bit more intuition. While a lot of category theory is well documented in a lot of textbooks, items like cartesian multicategories only really exist in research papers. Plus, you're drawing intuition from both PL and logic, which isn't always going to be clear to all readers.

I would actually be happy to provide such documentation - but I'd need access to your fork to do so.

Reply:
I'll give you access anyway, but also set about documenting stuff a bit this weekend. If I don't do it this weekend, that'll be a cue that I need some help.